mateweather/mateweather-dialog.c

```diff
@@ -502,6 +502,49 @@ static PangoFontDescription* get_system_monospace_font(void)
     return desc;
 }
 
+static void
+override_widget_font (GtkWidget            *widget,
+                      PangoFontDescription *font)
+{
+    GtkCssProvider *provider;
+    gchar          *css;
+    gchar          *family;
+    gchar          *weight;
+    const gchar    *style;
+    gchar          *size;
+
+    family = g_strdup_printf ("font-family: %s;", pango_font_description_get_family (font));
+
+    weight = g_strdup_printf ("font-weight: %d;", pango_font_description_get_weight (font));
+
+    if (pango_font_description_get_style (font) == PANGO_STYLE_NORMAL)
+        style = "font-style: normal;";
+    else if (pango_font_description_get_style (font) == PANGO_STYLE_ITALIC)
+        style = "font-style: italic;";
+    else
+        style = "font-style: oblique;";
+
+    size = g_strdup_printf ("font-size: %d", (pango_font_description_get_size (font) / PANGO_SCALE));
+
+    if (pango_font_description_get_size_is_absolute (font))
+        size = g_strconcat (size, "px;", NULL);
```

Comment:
Unfortunately you have introduced memory leak! `size` allocated by `g_strdup_printf` must be freed, but you overwrite size with new pointer...

Why not something like this?:
```
size = g_strdup_printf ("font-size: %d%s;",
                        pango_font_description_get_size (font) / PANGO_SCALE,
                        pango_font_description_get_size_is_absolute (font) ? "px" : "pt");
```

Reply:
thanks!